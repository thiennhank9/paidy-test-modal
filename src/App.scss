.app {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.bottom {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 100px;
}

.term-line {
  padding: 10px 0;
  font-size: 13px;
}
