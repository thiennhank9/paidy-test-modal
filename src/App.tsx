/* eslint-disable jsx-a11y/anchor-is-valid */
import React from "react";
import {
  Input,
  Label,
  Link,
  Button,
  Modal,
  Logo,
  InputPhone
} from "./components";
import "./App.scss";

function App() {
  return (
    <div className="app">
      <Modal>
        <Modal.Header>
          <section className="header">
            <div>
              <h1>SHOP NAME</h1>
              <h1>¥ 99,999</h1>
            </div>
            <Logo src="paidy-logo.png" alt="paidy-logo" />
          </section>
        </Modal.Header>
        <Modal.Body>
          <form>
            <Label>メールアドレス</Label>
            <Input />
            <Label>携帯電話番号</Label>
            <InputPhone />
            <div className="bottom">
              <span className="term-line">
                <Link>利用規約・個人情報取扱条項</Link> に同意して
              </span>
              <Button>次へ</Button>
            </div>
          </form>
        </Modal.Body>
      </Modal>
    </div>
  );
}

export default App;
