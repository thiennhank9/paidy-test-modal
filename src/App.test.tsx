import TestRenderer from "react-test-renderer";
import App from "./App";

describe("App", () => {
  test("render App", () => {
    const testRenderer = TestRenderer.create(<App />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
    const testInstance = testRenderer.root;
    expect(testInstance.findAllByType("input").length).toBe(2);
    expect(testInstance.findByType("button")).toBeDefined();
  });
});