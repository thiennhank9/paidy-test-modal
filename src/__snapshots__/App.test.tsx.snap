// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render App 1`] = `
<div
  className="app"
>
  <div
    className="modal"
  >
    <div
      className="modal-header"
    >
      <section
        className="header"
      >
        <div>
          <h1>
            SHOP NAME
          </h1>
          <h1>
            ¥ 99,999
          </h1>
        </div>
        <img
          alt="paidy-logo"
          className="logo"
          src="paidy-logo.png"
        />
      </section>
    </div>
    <div
      className="modal-body"
    >
      <form>
        <label>
          メールアドレス
        </label>
        <input />
        <label>
          携帯電話番号
        </label>
        <input
          aria-label="input-phone"
          onKeyPress={[Function]}
          onPaste={[Function]}
        />
        <div
          className="bottom"
        >
          <span
            className="term-line"
          >
            <a>
              利用規約・個人情報取扱条項
            </a>
             に同意して
          </span>
          <button>
            次へ
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
